test/e2e/azure_test.go

```diff
@@ -186,6 +186,16 @@ var _ = Describe("Workload cluster creation", func() {
 					},
 				}, result)
 
+				By("Verifying that expected Daemonset Pods are running", func() {
```

Comment:
Can we do this with less code duplication?

Reply:
> > @CecileRobertMichon do you prefer this updated use of the PostMachinesProvisioned func?
> 
> yes 100%. Although not sure why we need to pass in a func as param, commented on that.
> 
> @marosset has a good point, there is a lot of duplication. Perhaps we could refactor our uses of the clusterctl `ApplyClusterTemplateAndWait` with a wrapper helper func that adds all the common stuff like this by default and just takes the stuff that changes as parameters. That's probably out of scope here but would be a nice code cleanup.

Maybe just leave a TODO comment?
I'm OK doing the cleanup later

Reply:
Let's open an issue (TODO comments tend to get lost)